internal/transport/http_util_test.go

```diff
@@ -215,6 +218,37 @@ func (s) TestParseDialTarget(t *testing.T) {
 	}
 }
 
+type badNetworkConn struct {
+	net.Conn
+}
+
+func (c *badNetworkConn) Write([]byte) (int, error) {
+	return 0, io.EOF
+}
+
+func TestWriteBadConnection(t *testing.T) {
+	data := []byte("test_data")
+	writeBufferSize := len(data) - 1
```

Comment:
I believe this should be `(len(data) - 1) / 2`. Reason being that `w.flushKeepBuffer()` resets `w.offset` to 0 before returning the error. So `w.Write()` is able to able to copy `w.batchSize` more bytes from the input data.
The test presently doesn't timeout when the changes made in `http_util.go` are reverted.